src/events/processSignatureRequests.js

```diff
@@ -16,23 +16,18 @@ const { MAX_CONCURRENT_EVENTS } = require('../utils/constants')
 const { VALIDATOR_ADDRESS, VALIDATOR_ADDRESS_PRIVATE_KEY } = process.env
 const { HttpListProviderError } = HttpListProvider
 
-async function estimateGas(web3, bridgeContract, signature, message, address) {
+async function estimateGas({ web3, homeBridge, validatorContract, signature, message, address }) {
   try {
-    const gasEstimate = await bridgeContract.methods
-      .submitSignature(signature, message)
-      .estimateGas({
-        from: address
-      })
+    const gasEstimate = await homeBridge.methods.submitSignature(signature, message).estimateGas({
+      from: address
+    })
     return gasEstimate
   } catch (e) {
     if (e instanceof HttpListProviderError) {
       throw e
     }
 
     // Check if address is validator
-    const validatorContractAddress = await bridgeContract.methods.validatorContract().call()
-    const validatorContract = new web3.eth.Contract(bridgeValidatorsABI, validatorContractAddress)
-
     const isValidator = await validatorContract.methods.isValidator(address).call()
 
     if (!isValidator) {
```

Comment:
Yes, probably it is good idea to stop the process at all if the signer is not a validator. Could you use a specific exit code in order to we could distinguish different reasons why the process was stopped.
If it will be done in such manner it makes sense to forbid to run the watcher at all if the private key used to sign transactions does not belong to any address from the validators list.

Reply:
> Could you use a specific exit code in order to we could distinguish different reasons why the process was stopped.

Sure.

> If it will be done in such manner it makes sense to forbid to run the watcher at all if the private key used to sign transactions does not belong to any address from the validators list.

You mean checking this during the initialization of the watcher? I totally agree with that.

Reply:
> You mean checking this during the initialization of the watcher? I totally agree with that.
Yes